controllers/gitrepository_controller.go

```diff
@@ -459,13 +459,21 @@ func (r *GitRepositoryReconciler) reconcileSource(ctx context.Context,
 			// Return error as the world as observed may change
 			return sreconcile.ResultEmpty, e
 		}
+		authData = secret.Data
+	}
 
-		// Configure strategy with secret
-		authOpts, err = git.AuthOptionsFromSecret(obj.Spec.URL, &secret)
-	} else {
-		// Set the minimal auth options for valid transport.
-		authOpts, err = git.AuthOptionsWithoutSecret(obj.Spec.URL)
+	u, err := url.Parse(obj.Spec.URL)
+	if err != nil {
+		e := serror.NewStalling(
+			fmt.Errorf("failed to parse url '%s': %w", obj.Spec.URL, err),
+			sourcev1.URLInvalidReason,
+		)
+		conditions.MarkTrue(obj, sourcev1.FetchFailedCondition, e.Reason, e.Err.Error())
+		return sreconcile.ResultEmpty, e
 	}
+
+	// Configure authentication strategy to access the source
+	authOpts, err := git.NewAuthOptions(*u, authData)
```

Comment:
The previous (newly added, removed in the recent commit) error handling code may have been better for this as this operation seems to have nothing to do with the git implementation which is highlighted in the error below.